packages/waku/src/router/client.ts

```diff
@@ -301,8 +305,15 @@ const InnerRouter = ({ routerData }: { routerData: RouterData }) => {
   const refetch = useRefetch();
 
   const [route, setRoute] = useState(() =>
-    parseRoute(new URL(window.location.href)),
+    parseRoute(new URL(window.location.href), true),
   );
+
+  // Update the route post-load to include the current hash.
+  useEffect(() => {
+    setRoute(parseRoute(new URL(window.location.href)));
```

Comment:
Do we not `ignoreHash` here?

Reply:
No, thats where we want to get it, as the comment states.

Reply:
I left a note in there.